### DISCUSSION TITLE
TypeError: r is not a function

### DISCUSSION THREAD

**Discussion by throne1986** [prevwong/craft.js#285] [Q&A]
Hi everyone,

I want to run the landing example using pure react js without typescript, so I have refactored the given example in this repo,

Here is my new repo: https://github.com/throne1986/landing-app

But, unfortunately, I keep getting the error: `TypeError: r is not a function

![127243905-c11b3d98-1360-4273-890e-ca68752ea058 1](https://user-images.githubusercontent.com/27904724/127295845-94d4a719-59c3-4a69-8960-a11f48f7ed45.png)

I am out of idea what I am doing wrong, can someone please tell me what is wrong here?

Thanks

**Comment by ankri** [comment]
When I follow your [index.js](https://github.com/throne1986/landing-app/blob/master/src/index.js#L12) and your [App.js](https://github.com/throne1986/landing-app/blob/master/src/App.js#L9) then the [Viewport.js](https://github.com/throne1986/landing-app/blob/master/src/features/editor/Viewport/index.js)

You need to wrap your `<Viewport />` with a `<Editor />` component from `@craft.js/core` or remove the use of [`useEditor`](https://github.com/throne1986/landing-app/blob/master/src/features/editor/Viewport/index.js#L11-L13) since you are first using the `<Editor />` in your [Page.js](https://github.com/throne1986/landing-app/blob/master/src/features/pages/index.js#L32)

**Comment by ankri** [comment]
In other words, you cannot use `useEditor` outside the context of the `<Editor />` component from the core package. 

I will try to open a PR that will produce a clearer error message for those cases.

**Comment by throne1986** [comment]
> When I follow your [index.js](https://github.com/throne1986/landing-app/blob/master/src/index.js#L12) and your [App.js](https://github.com/throne1986/landing-app/blob/master/src/App.js#L9) then the [Viewport.js](https://github.com/throne1986/landing-app/blob/master/src/features/editor/Viewport/index.js)
> 
> You need to wrap your `<Viewport />` with a `<Editor />` component from `@craft.js/core` or remove the use of [`useEditor`](https://github.com/throne1986/landing-app/blob/master/src/features/editor/Viewport/index.js#L11-L13) since you are first using the `<Editor />` in your [Page.js](https://github.com/throne1986/landing-app/blob/master/src/features/pages/index.js#L32)

Thanks, bro that is true solved the error, but now I have another issue when I am rendering the Pages.js component it keeps rendering infinite , 

![image](https://user-images.githubusercontent.com/27904724/127315221-1ac04878-8366-4275-9484-77c318b5ca9e.png)




 [https://github.com/throne1986/landing-app/blob/master/src/features/editor/Viewport/index.js](url)
![image](https://user-images.githubusercontent.com/27904724/127306599-f57afadb-19ba-4f91-a59a-420d4700f1f1.png)

I have updated my repo, please check what is wrong here

**Comment by ankri** [comment]
You are rendering `<Viewport />` again in [`<Pages />`](https://github.com/throne1986/landing-app/blob/master/src/features/pages/index.js#L48) which will render `<Pages />` again. And so on.

**Comment by ankri** [comment]
Since this is not a craft related issue, we can maybe close this issue and move further communication to your thread in the [discussions](https://github.com/prevwong/craft.js/discussions/283)?

**Comment by throne1986** [comment]
> Since this is not a craft related issue, we can maybe close this issue and move further communication to your thread in the [discussions](https://github.com/prevwong/craft.js/discussions/283)?

ok u can transfer to discussion
